### ISSUE TITLE
Choking on ligature, I think - gets read as �

### ISSUE THREAD

**Issue by renanbirck** [pymupdf/PyMuPDF#2876]
## Describe the bug (mandatory)
I'm trying to parse [this PDF]( https://www.joinville.sc.gov.br/wp-content/uploads/2023/11/Pesquisa-de-Precos-Combustiveis-novembro-2023.pdf). However, the first line of the first page gets read as 'Pesquisa de Preços - Combus�veis', whith the � instead of the 'tí' that should be there.

Other PDF parsing libraries, like `pdfium`, don't have this problem. 

## To Reproduce (mandatory)
A minimal example of code that causes the problem is:

```python
import fitz
doc = fitz.open('pesquisa-de-Precos-Combustiveis-novembro-2023.pdf')
doc[0].get_text().split('\n')[0]

```
and in the output there is an �

## Expected behavior (optional)
The correct output (the `tí` not the �).

## Your configuration (mandatory)
 - Fedora 39 64-bit
 - Python 3.12 64-bit
 - PyMuPDF 1.23.7, installed from wheel

**Comment by JorjMcKie** [comment]
An example glyph (number 417) is wrapped in structure information containing the unfolded / intended text:
```xml
<structure standard="Span">
    <metatext type="actualtext" txt="tí">
        <fill_text colorspace="DeviceRGB" color="1 1 1" ri="1" bp="1" op="0" opm="0" transform=".75 0 0 .75 .75 3">
            <span font="BFWFQN+Calibri-Bold" wmode="0" bidi="0" trm="28 0 0 -28">
                <g unicode="�" glyph="417" x="365.64698" y="50" adv=".5854492"/>
            </span>
        </fill_text>
    </metatext>
</structure>
```
We will forward this to MuPDF.

**Comment by JorjMcKie** [comment]
Turns out that there already exists a support request in MuPDF's system [here](https://bugs.ghostscript.com/show_bug.cgi?id=702966).

**Comment by JorjMcKie** [comment]
This issue is resolved by version 1.23.14.
